frontend/src/features/admin-form/settings/components/MrfFormEmailSection.tsx

```diff
@@ -0,0 +1,205 @@
+    },
+    [checkIsEmail],
+  )
+
+  const { stepsToNotify, emails } = settings
+
+  const {
+    handleSubmit,
+    control,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm<{
+    [WorkflowEmailMultiSelectName]: string[]
+    [OtherPartiesEmailInputName]: string[]
+  }>({
+    defaultValues: {
+      [WorkflowEmailMultiSelectName]: stepsToNotify,
+      [OtherPartiesEmailInputName]: emails,
+    },
+  })
+
+  const { mutateMrfEmailNotifications } = useMutateFormSettings()
+
+  const handleSubmitEmailNotificationSettings = useCallback(
+    ({ nextStaticEmails, nextStepsToNotify }) => {
+      if (
+        isEqual(nextStaticEmails, emails) &&
+        isEqual(nextStepsToNotify, stepsToNotify)
+      ) {
+        return
+      }
+      return mutateMrfEmailNotifications.mutate({
+        emails: nextStaticEmails,
+        stepsToNotify: nextStepsToNotify,
+      })
+    },
+    [mutateMrfEmailNotifications, emails, stepsToNotify],
+  )
+  const DEBOUNCE_DELAY_IN_MS = 800
+  const onSubmit = useCallback(
+    (formData) => {
+      const selectedSteps = formData[WorkflowEmailMultiSelectName]
+      const selectedEmails = formData[OtherPartiesEmailInputName]
+
+      return handleSubmitEmailNotificationSettings({
+        nextStepsToNotify: selectedSteps,
+        nextStaticEmails: selectedEmails,
+      })
+    },
+    [handleSubmitEmailNotificationSettings],
+  )
+  const onSubmitDebounced = useCallback(
+    (formData) => debounce(() => onSubmit(formData), DEBOUNCE_DELAY_IN_MS)(),
+    [onSubmit],
+  )
+
+  const handleWorkflowEmailMultiSelectChange = useCallback(() => {
+    handleSubmit(onSubmitDebounced)()
+  }, [handleSubmit, onSubmitDebounced])
+
+  const handleOtherPartiesEmailInputBlur = useCallback(() => {
+    const uniqueValidEmails = uniq(
+      filterInvalidEmails(getValues(OtherPartiesEmailInputName)),
+    )
+    setValue(OtherPartiesEmailInputName, uniqueValidEmails)
+    handleSubmit(onSubmit)()
+  }, [getValues, handleSubmit, onSubmit, setValue, filterInvalidEmails])
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Box>
+        <FormLabel>Notify respondents in your workflow</FormLabel>
+        <Skeleton isLoaded={!isLoading}>
+          <Box my="0.75rem">
+            <Controller
+              control={control}
+              name={WorkflowEmailMultiSelectName}
+              render={({
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                field: { value: values = [], onChange, ...rest },
+              }) => (
+                <MultiSelect
+                  items={formWorkflowStepsWithStepNumber.map((step) => ({
+                    label: `Respondent(s) in Step ${step.stepNumber}`,
+                    value: step._id,
+                  }))}
+                  values={values}
+                  onChange={(values) => {
+                    onChange(values)
+                    handleWorkflowEmailMultiSelectChange()
```

Comment:
This section feels a little too complex to handle. Looking at seeing how this can be simplified.

Is the reason for this additional function call + debounce due to the fact that we don't have an `onBlur` to trigger the event?

Reply:
hey ken, im trying to implement the onBlur logic for the form multiselect component, and i realised why the MultiSelect component may not have had an `onBlur` prop implemented previously

ive set an onBlur callback function which is invoked when the focused input box is blurred. this works fine for when selecting elements from the dropdown.

However, when clearing elements, for example when clicking the x button, the input box is not focused ever and hence the onBlur event does not trigger. This means that the onBlur callback is not invoked

<img width="589" alt="image" src="https://github.com/user-attachments/assets/7388c4d1-da53-4fcb-bf8d-7c09a97ba9a0">
element is focused when selecting steps from the virtual list 

however, when clearing, input component is never focused 
<img width="586" alt="image" src="https://github.com/user-attachments/assets/f7127649-f34e-436f-8c1a-5a7855200d7b">
hence, admin must click on input box to focus then click out to trigger `onBlur` callback. 

Conclusion: use debounce + `onChange`

Reply:
Adding the following comment for documentation of what we discussed: 

Kevin: 
there are 3 approaches i can take:
keep current implementation in GH PR: use onChange for everything (current design) -> which is imo easier to understand. dont think it will negatively affect admin's usage of the toggle
i can use onBlur for when user adds steps and onChangeonly when the user clicks on xbutton. but there are some tradeoffs discovered only after I tried implementing:
still better to have debounce for x to avoid multiple api calls when user deletes multiple steps, hence does not solve problem from 1
different behavior for adding steps (updates only when on blur) vs deleting steps (updates immediately)
need to add a if (selectedItems.length < prevSelectedItems.length) submitForm;
need to call submit form twice
implement clicking x button focuses onto the input box. but this is indeed a strange behavior
my take is since the intention is to make the code more understandable, options 2 and 3 dont really achieve that.
I do feel that approach 1 is already reasonably understandable and we might not need to change to 2 or 3.
current implementation is probably acceptable for current use case and i should probably move on to next part instead of spending more time on this.

Ken: 
Hm i had troubles understanding 1.
i.e., why is this here
```onChange={(values) => {
    onChange(values)
    handleWorkflowEmailMultiSelectChange()
}}```
Then we have debounce etc to add in the complexity. When you think of using a MultiCombobox, do you have to think of debounce + onChange to track every changes, or you think that MultiCombobox should tell you (1) something changed or (2) when the user had done updating? They are have two different levels of functionality.
That’s one of the main reason i asked, why was onChange used. Was it due to convenience or restriction. It sounded like more like a convenience rather than restriction.
Natural behaviour of automatic fields updates on FormSG is generally onBlur (and as well as other projects that i’ve worked on).
You can go think about it. If approach (1) is still the best, then lets take that.

Reply:
Okay, this is resolved! 

I've used option 3 which is i think the cleanest solution and upon trying it out it does look intuitive and not really odd to focus on remove and avoids the delay issue faced with debouncing.